source/common/common/thread.h

```diff
@@ -190,5 +190,15 @@ struct MainThread {
   std::thread::id main_thread_id_{std::this_thread::get_id()};
 };
 
+#define TRY_ASSERT_MAIN_THREAD                                                                     \
```

Comment:
Can you add documentation on this in terms of why it's here, etc.

Reply:
Thanks, will do.

Reply:
Thanks, comments added.